File: game-price-tracker-be/tests/domain/currency.test.ts
import { Currency } from '../../src/game-price-tracker-be/domain/value_object/currency';

describe('Currency VO', () => {
  it('should throw an error if amount is not positive', () => {
    expect(() => new Currency(0, 'USD')).toThrow('Amount must be positive.');
  });

  it('should throw an error if amount is negative', () => {
    expect(() => new Currency(-10, 'USD')).toThrow('Amount must be positive.');
  });

  it('should be valid with positive amount', () => {
    const currency = new Currency(100, 'USD');
    expect(currency.isValid()).toBe(true);
  });
});

File: game-price-tracker-be/core/src/test/kotlin/com/gamepricetracker/core/domain/models/CurrencyTest.kt
package com.gamepricetracker.core.domain.models

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class CurrencyTest {

    @Test
    fun `throws when amount is zero`() {
        val ex = shouldThrow<IllegalArgumentException> {
            Currency(amount = 0.0, code = "USD")
        }
        ex.message shouldBe "Amount must be positive."
    }

    @Test
    fun `throws when amount is negative`() {
        val ex = shouldThrow<IllegalArgumentException> {
            Currency(amount = -10.0, code = "USD")
        }
        ex.message shouldBe "Amount must be positive."
    }

    @Test
    fun `is valid when amount is positive`() {
        val currency = Currency(amount = 100.0, code = "USD")
        currency.isValid() shouldBe true
    }
}
